widget/widget_renderer.go

```diff
@@ -0,0 +1,34 @@
+package widget
+
+import (
+	"fyne.io/fyne"
+	"fyne.io/fyne/theme"
+)
+
+// When using the baseRenderer the embedding renderer should call
+// layoutShadow(contentSize, contentPos) to lay out the shadow.
+type baseRenderer struct {
+	objects []fyne.CanvasObject
+	sh      fyne.CanvasObject
+}
+
+func newBaseRenderer(objects []fyne.CanvasObject, level int) *baseRenderer {
+	var shadow fyne.CanvasObject
+	if level > 0 {
+		shadow = newShadow(shadowAround, level*theme.Padding()/2)
+		objects = append(objects, shadow)
+	}
+	return &baseRenderer{objects, shadow}
+}
+
+func (r *baseRenderer) Objects() []fyne.CanvasObject {
```

Comment:
Can baseRenderer also implement BackgroundColor and Destroy since most of these implementations are "return theme.BackgroundColor()" and empty respectively?

Reply:
I agree that these two  are good candidates for the base renderer. No I think that I chose the wrong name and the renderer should be named `shadowingBaseRenderer` or alike. The simple implementation of common methods would then go into the `baseRenderer` which will be embedded by the `shadowingBaseRenderer`.
WDYT?